### PR TITLE
fix(flex/space/size): fixes nested styles when using flex/space/size on react-native

### PR DESCRIPTION
**Pull request by shubhnik**
fixes nested styles when using flex/space/size and add test cases for the same.

### CODE REVIEW

**Comment by shubhnik** [comment]
I need to write tests wherein it tests how `Flex/Space`, `Flex/Size` etc plays together. Currently, I have written those tests in `flex.native.test.js`. What do you guys suggest where should I place tests like these, kind of integration tests?... @shridhar52war @deorukhkarchaitanya @kamleshchandnani

File: src/atoms/Flex/__tests__/Flex.native.test.js [hunk 1]
```diff
@@ -1,15 +1,140 @@
...
+import View from '../../View';
 import Flex from '../index';
+import Space from '../../Space';
+import Size from '../../Size';
+
+const defaultFlexStyles = { flexDirection: 'column' };
+const defaultSpaceStyles = {};
+const defaultSizeStyles = {};
 
 describe('Renders <Flex /> correctly', () => {
```
Comment (shridhar52war) [inline src/atoms/Flex/__tests__/Flex.native.test.js]:
```suggestion
describe('<Flex />', () => {
```

File: src/atoms/Flex/__tests__/Flex.native.test.js [hunk 2]
```diff
@@ -1,15 +1,140 @@
...
-  it('snapshot testing', () => {
-    const { container } = render(
+  it('renders child of <Flex />(no styles) correctly', () => {
```
Comment (shridhar52war) [inline src/atoms/Flex/__tests__/Flex.native.test.js]: You don't have to use <Flex /> everytime as `<Flex />` as already in `describe("<Flex />")`

File: src/atoms/Flex/__tests__/Flex.native.test.js [hunk 3]
```diff
@@ -1,15 +1,140 @@
...
+        <View testID={'flex-view'} />
+      </Flex>,
+    );
+    const childView = getByTestId('flex-view');
+    expect(childView.props.style).toMatchObject([defaultFlexStyles]);
+    expect(container).toMatchSnapshot();
+  });
+});
+
+describe('<Flex /> and <Space /> together plays properly', () => {
```
Comment (shridhar52war) [inline src/atoms/Flex/__tests__/Flex.native.test.js]:
```suggestion
describe('<Flex /> and <Space />', () => {
```

File: src/atoms/Flex/__tests__/Flex.native.test.js [hunk 4]
```diff
@@ -1,15 +1,140 @@
...
+  it('renders child of <Flex />(no styles) and <Space />(no styles) correctly', () => {
```
Comment (shridhar52war) [inline src/atoms/Flex/__tests__/Flex.native.test.js]:
```suggestion
  it('renders <Space /> as child component of <Flex />with expected props', () => {
```

File: src/atoms/Flex/__tests__/Flex.native.test.js [hunk 5]
```diff
@@ -1,15 +1,140 @@
...
+          <View testID={'flex-space-view'} />
+        </Space>
+      </Flex>,
+    );
+    const childView = getByTestId('flex-space-view');
+    expect(childView.props.style).toMatchObject([defaultSpaceStyles, defaultFlexStyles]);
+    expect(container).toMatchSnapshot();
+  });
+
+  it('renders child of <Space />(no styles) and <Flex />(no styles) correctly', () => {
```
Comment (shridhar52war) [inline src/atoms/Flex/__tests__/Flex.native.test.js]:
```suggestion
  it('renders <Flex /> as child component of <Space />', () => {
```

File: src/atoms/Flex/__tests__/Flex.native.test.js [hunk 6]
```diff
@@ -1,15 +1,140 @@
...
+        </Flex>
+      </Space>,
+    );
+    const childView = getByTestId('space-flex-view');
+    expect(childView.props.style).toMatchObject([defaultFlexStyles, defaultSpaceStyles]);
+    expect(container).toMatchSnapshot();
+  });
+});
+
+describe('<Flex /> and <Size /> together plays properly', () => {
```
Comment (shridhar52war) [inline src/atoms/Flex/__tests__/Flex.native.test.js]:
```suggestion
describe('<Flex /> and <Size />', () => {
```

File: src/atoms/Flex/__tests__/Flex.native.test.js [hunk 7]
```diff
@@ -1,15 +1,140 @@
...
+  it('renders child of <Flex />(no styles) and <Size />(no styles) correctly', () => {
```
Comment (shridhar52war) [inline src/atoms/Flex/__tests__/Flex.native.test.js]:
```suggestion
  it('renders <Size /> as child component of <Flex />', () => {
```

File: src/atoms/Flex/__tests__/Flex.native.test.js [hunk 8]
```diff
@@ -1,15 +1,140 @@
...
+          <View testID={'flex-size-view'} />
+        </Size>
+      </Flex>,
+    );
+    const childView = getByTestId('flex-size-view');
+    expect(childView.props.style).toMatchObject([defaultSpaceStyles, defaultFlexStyles]);
+    expect(container).toMatchSnapshot();
+  });
+
+  it('renders child of <Size />(no styles) and <Flex />(no styles) correctly', () => {
```
Comment (shridhar52war) [inline src/atoms/Flex/__tests__/Flex.native.test.js]:
```suggestion
  it('renders <Flex /> as child component of <Size />', () => {
```

File: src/atoms/Flex/__tests__/Flex.native.test.js [hunk 9]
```diff
@@ -1,15 +1,140 @@
...
+        </Flex>
+      </Size>,
+    );
+    const childView = getByTestId('size-flex-view');
+    expect(childView.props.style).toMatchObject([defaultFlexStyles, defaultSpaceStyles]);
+    expect(container).toMatchSnapshot();
+  });
+});
+
+describe('<Flex />, <Size />, <Space /> together plays properly', () => {
```
Comment (shridhar52war) [inline src/atoms/Flex/__tests__/Flex.native.test.js]: Rename test suite and test cases as per above comments

File: src/atoms/Flex/__tests__/Flex.native.test.js (hunk 1 above)
Comment (shubhnik) [reply src/atoms/Flex/__tests__/Flex.native.test.js]: Let's sit and discuss conventions.

File: src/atoms/Flex/__tests__/Flex.native.test.js (hunk 8 above)
Comment (shubhnik) [reply src/atoms/Flex/__tests__/Flex.native.test.js]: Not just Flex as child of Size but it mainly tests the **View** which is a child of Flex which in turn is a child of Size has **received correct styles**.

File: src/atoms/Flex/__tests__/Flex.native.test.js (hunk 7 above)
Comment (shubhnik) [reply src/atoms/Flex/__tests__/Flex.native.test.js]: Not just Flex as child of Size but it mainly tests the View which is a child of Flex which in turn is a child of Size has received correct styles.

File: src/atoms/Flex/__tests__/Flex.native.test.js (hunk 4 above)
Comment (shubhnik) [reply src/atoms/Flex/__tests__/Flex.native.test.js]: Not just Flex as child of Space but it mainly tests the **View** which is a child of Flex which in turn is a child of Space has **received correct styles**.

Comment (shubhnik) [reply src/atoms/Flex/__tests__/Flex.native.test.js]: Not just Flex as child of Size but it mainly tests the **View** which is a child of Flex which in turn is a child of Size has **received correct styles**.

File: src/atoms/Flex/__tests__/Flex.native.test.js (hunk 3 above)
Comment (shubhnik) [reply src/atoms/Flex/__tests__/Flex.native.test.js]: Trying to be as descriptive and that also using layman language.

File: src/atoms/Flex/Flex.native.js [hunk 10]
```diff
@@ -19,7 +19,12 @@ const Flex = styled(
...
     style,
     ...props
-  }) => React.cloneElement(children, { ...props, style: [style, children.props.style] }),
+  }) => {
+    return React.cloneElement(children, {
+      ...props,
+      style: [...style, children.props.style].filter(Boolean),
+    });
+  },
 )`
```
Comment (deorukhkarchaitanya) [inline src/atoms/Flex/Flex.native.js]: Can't we just do this?

```suggestion
  }) => React.cloneElement(children, { ...props, style: [style, children.props.style].filter(Boolean) }),
```

**Pull request merged**